bindings/java/src/org/sleuthkit/datamodel/blackboardutils/attributes/TskGeoTrackpointsUtil.java

```diff
@@ -0,0 +1,342 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.datamodel.blackboardutils.attributes;
+
+import com.google.gson.Gson;
+import com.google.gson.annotations.SerializedName;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.sleuthkit.datamodel.BlackboardAttribute;
+import org.sleuthkit.datamodel.blackboardutils.attributes.TskGeoTrackpointsUtil.GeoTrackPointList;
+import org.sleuthkit.datamodel.blackboardutils.attributes.TskGeoTrackpointsUtil.GeoTrackPointList.GeoTrackPoint;
+
+/**
+ * Utility class for Translating TSK_GEO_TRACKPOINTS attribute values to
+ * GeoTrackPointList objects and GeoTrackPointList to BlackboardAttributes.
+ */
+public final class TskGeoTrackpointsUtil implements BlackboardAttributeTranslator<GeoTrackPointList> {
+
+	private final String moduleName;
+
+	/**
+	 * Constructs a new instance of the Translator Utility.
+	 *
+	 * @param moduleName Name of calling module.
+	 */
+	public TskGeoTrackpointsUtil(String moduleName) {
+		this.moduleName = moduleName;
+	}
+
+	@Override
+	public BlackboardAttribute toAttribute(GeoTrackPointList value) {
+
+		if (value == null) {
+			throw new IllegalArgumentException("toAttribute was pass a null list.");
+		}
+
+		return new BlackboardAttribute(
+				BlackboardAttribute.ATTRIBUTE_TYPE.TSK_GEO_TRACKPOINTS,
+				moduleName,
+				toJSON(value));
+	}
+
+	@Override
+	public GeoTrackPointList fromAttribute(BlackboardAttribute attribute) {
+		if (attribute == null) {
+			throw new IllegalArgumentException("Null value passed to fromAttribute");
+		}
+
+		BlackboardAttribute.ATTRIBUTE_TYPE type = BlackboardAttribute.ATTRIBUTE_TYPE.fromID(attribute.getAttributeType().getTypeID());
+		if (type != BlackboardAttribute.ATTRIBUTE_TYPE.TSK_GEO_TRACKPOINTS) {
+			throw new IllegalArgumentException(String.format("Invalid attribute of type %s passed to fromAttribute method. Attribute of type TSK_GEO_TRACKPOINTS is required", type.getDisplayName()));
+		}
+
+		return fromJSON(attribute.getValueString());
+	}
+
+	/**
+	 * Creates a GeoTrackPointList from the given JSON string.
+	 *
+	 * @param jsonString JSon string of track points.
+	 *
+	 * @return	Timestamp ordered list of GeoTrackPoints, empty list will be
+	 *         returned if jsonString is null or empty.
+	 */
+	private static GeoTrackPointList fromJSON(String jsonString) {
+		if (jsonString == null || jsonString.isEmpty()) {
+			return null;
+		}
+
+		return (new Gson()).fromJson(jsonString, GeoTrackPointList.class);
+	}
+
+	/**
+	 * Returns a JSON string representing the given object.
+	 *
+	 * @return JSON string
+	 */
+	private static String toJSON(GeoTrackPointList pointList) {
+		Gson gson = new Gson();
+		return gson.toJson(pointList);
+	}
+
+	/**
+	 * Helper class to make it easier to serialize and deserialize the
+	 * TSK_GEO_TRACKPOINTS attribute.
```

Comment:
This is not the view we are exposing to clients. The JSON is an implementation detail. It would be better to document this public class as a list of track points, rather than in programming terms.